test/stripe/payment_intent_test.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require File.expand_path("../../test_helper", __FILE__)
+
+module Stripe
+  TEST_RESOURCE_ID = "pi_123".freeze
+
+  class PaymentIntentTest < Test::Unit::TestCase
+    should "be listable" do
+      stub_request(:get, "#{Stripe.api_base}/v1/payment_intents")
+        .to_return(body: JSON.generate(object: "list", data: [{ id: "pi_123", object: "payment_intent" }]))
+
+      payment_intents = Stripe::PaymentIntent.list
+      assert_requested :get, "#{Stripe.api_base}/v1/payment_intents"
+      assert payment_intents.data.is_a?(Array)
+      assert payment_intents.data[0].is_a?(Stripe::PaymentIntent)
+    end
+
+    should "be retrievable" do
+      stub_request(:get, "#{Stripe.api_base}/v1/payment_intents/pi_123")
+        .to_return(body: JSON.generate(id: "pi_123", object: "payment_intent"))
+
+      payment_intent = Stripe::PaymentIntent.retrieve("pi_123")
+      assert_requested :get, "#{Stripe.api_base}/v1/payment_intents/pi_123"
+      assert payment_intent.is_a?(Stripe::PaymentIntent)
+    end
+
+    should "be creatable" do
+      stub_request(:post, "#{Stripe.api_base}/v1/payment_intents")
+        .to_return(body: JSON.generate(id: "pi_123", object: "payment_intent"))
+
+      payment_intent = Stripe::PaymentIntent.create
+      assert_requested :post, "#{Stripe.api_base}/v1/payment_intents"
+      assert payment_intent.is_a?(Stripe::PaymentIntent)
+    end
+
+    should "be saveable" do
+      stub_request(:get, "#{Stripe.api_base}/v1/payment_intents/pi_123")
+        .to_return(body: JSON.generate(id: "pi_123", object: "payment_intent", metadata: {}))
```

Comment:
When testing an instance method, instead of stubbing the retrieve request then initializing the resource with `.retrieve`, I'd recommend just creating the resource directly with `.new`. This way only one request is sent which can clarify errors if `assert_requested` fails.

Reply:
I mostly mimicked what is in other tests though but will fix